src/test/java/org/cbioportal/cdd/ClinicalDataDictionaryTest.java

```diff
@@ -95,6 +99,41 @@ public void getClinicalAttributeMetadataFilteredTest() throws Exception {
         assertThat(response.getBody(), containsString("{\"column_header\":\"AGE\",\"display_name\":\"Diagnosis Age\",\"description\":\"Age at which a condition or disease was first diagnosed.\",\"datatype\":\"NUMBER\",\"attribute_type\":\"PATIENT\",\"priority\":\"1\"}"));
     }
 
+    @Test
+    public void validateCacheTest() throws Exception {
+        Calendar currentDate = Calendar.getInstance();
+
+        // test condition where cache is not stale
+        // set cache age to a valid age (current date - MAXIMUM_CACHE_AGE_IN_DAYS + 1)
+        // cache is not stale so resetCache should not be called -- cache age should remain the smae
+        currentDate.add(Calendar.DATE, -(clinicalAttributesCache.MAXIMUM_CACHE_AGE_IN_DAYS - 1));
+        Date validDateOfLastCacheRefresh = currentDate.getTime();
+        clinicalAttributesCache.setDateOfLastCacheRefresh(validDateOfLastCacheRefresh);
+        if (clinicalAttributesCache.cacheIsStale()) {
+            clinicalAttributesCache.resetCache();
+        }
+        assertThat(clinicalAttributesCache.getDateOfLastCacheRefresh().toString(), equalTo(validDateOfLastCacheRefresh.toString()));
+ 
+        // test condition where cache is stale
+        // set cache age to a valid age (current date - MAXIMUM_CACHE_AGE_IN_DAYS - 3)
+        // cache is stale so resetCache should be called -- cache age should change
+        currentDate.add(Calendar.DATE, -(clinicalAttributesCache.MAXIMUM_CACHE_AGE_IN_DAYS + 3));
```

Comment:
Copy of a comment from analogous oncotree PR: 
here we are re-using the currentDate object. But wasn't current date offset already on line 109 to be two days before "now" ? So by subtracting 6 days on line 120 don't we get to a time point 8 days before "now"? This doesn't break what we want .. the second time point is "old enough", but if additional tests are added with new timepoints the programmer may be confused about why their times are not what they expect ... if I'm understanding Calendar properly.